software/script/chameleon_cli_unit.py

```diff
@@ -1679,30 +1681,188 @@ def on_exec(self, args: argparse.Namespace):
             'keep_rf_field': 0,
             'check_response_crc': 1,
         }
-        # TODO: auth first if a key is given
+
+        if param.pwd is not None:
+            options['keep_rf_field'] = 1
+            resp = self.cmd.hf14a_raw(options=options, resp_timeout_ms=200, data=struct.pack('!B', 0x1B)+param.pwd)
+            options['keep_rf_field'] = 0
+            options['auto_select'] = 0
+            print(f" - PACK: {resp[:2].hex()}")
```

Comment:
<img width="391" alt="Screenshot 2024-07-07 at 19 39 50" src="https://github.com/RfidResearchGroup/ChameleonUltra/assets/60693357/0c1cda2f-e3da-48ac-863e-a833882382b5">

it will

Reply:
Should we check NACK response?
![Xnip2024-07-08_00-50-48](https://github.com/RfidResearchGroup/ChameleonUltra/assets/2192626/7762ce07-b522-4156-a7ef-58743311ba51)

Reply:
Because I think to distinguish not found and wrong auth is important.

Reply:
I'll do. The cards I've tested with seem to just not send the response instead of sending NAKs

Reply:
Hm, that's a bit complicated. The `HF14A_RAW` command only returns the amount of bytes received, NAKs and ACKs get rounded to 1 byte. With this command we could just check if byte is 0xA0 or something else, but there should probably be another command that can return the amount of bits. Also for some reason while testing I've usually noticed an empty response returned instead of one byte

Reply:
I see, that's because when you want CRC calculation you can't receive ACKs and NAKs

Reply:
This should be done now.

<img width="354" alt="Screenshot 2024-07-08 at 01 05 17" src="https://github.com/RfidResearchGroup/ChameleonUltra/assets/60693357/79a8cb82-2b68-4200-8dc5-52f6c4336a07">

Reply:
how about try to set appendCrc to false and check if resp is `0x0A`?

ref: https://github.com/RfidResearchGroup/ChameleonUltra/blob/3cadd47eebbc719ab97626845373972756e3d80e/software/script/chameleon_cmd.py#L968

Reply:
It will be the PACK value instead of 0x0A on success. We can just add another command version that returns size in bits I think if we really want to ensure NAKs are NAKs.

Reply:
Just a recommendation. In my js sdk, I will set `wait_response` to true, `check_response_crc` to false, then check `pack + crc16` for success, NAK for auth failed (hf14a_raw return 8 bit is OK), and other for unknown response.

Reply:
NAK is `0x04`, not `0x0A`. My mistake.